### PR TITLE
Fix extension cluster submenu re-render (port #1996 to master)

### PR DESCRIPTION
**Pull request by jakolehm**
Port of #1996

### CODE REVIEW

**Review by nevalla — APPROVED** [review]

File: src/renderer/components/app.tsx [hunk 1]
```diff
@@ -143,22 +151,38 @@ export class App extends React.Component {
...
         }
       }
     });
+
+    Array.from(this.extensionRoutes.keys()).forEach((menu) => {
+      if (!rootItems.includes(menu)) {
+        this.extensionRoutes.delete(menu);
+      }
+    });
+  }
```
Comment (Nokel81) [inline src/renderer/components/app.tsx:182]: We shouldn't create an array just to iterate through it once and the throw it away.
```suggestion
    for (const meny of this.extensionRoutes.keys()) {
      if (!rootItems.includes(menu)) {
        this.extensionRoutes.delete(menu);
      }
    }
```

**Review by Nokel81 — APPROVED** [review]
Approving, though I don't know if you also want review commets (so I will leave some anyway).

File: src/renderer/components/app.tsx (hunk 1 above)
Comment (jakolehm) [reply src/renderer/components/app.tsx:182]: I'll create a separate PR about cleanup. I never remember that we can use `for` in these cases.

Comment (Nokel81) [reply src/renderer/components/app.tsx:182]: Ah I see, that is why `forEach` is so common. Though remember that `forEach` is one of the only "methods" on interators, so you don't need to convert to an array to use that method.

**Pull request merged**